node/tests/k8st/infra/build-operator.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+# This script produces a custom operator build that is used
+# in the ST and e2e tests in this repo.
+
+# Clone the repository if needed.
+set -e
+REPO=${REPO:-tigera/operator}
+BRANCH=${BRANCH:-master}
+
+rm -rf operator/
+echo "Cloning https://github.com/${REPO} @ ${BRANCH}"
+git clone --depth=1 https://github.com/${REPO} -b ${BRANCH} operator
+
+# Modify the versions that are in-use to match our locally built images.
+pushd operator
+make build/_output/bin/gen-versions
+build/_output/bin/gen-versions -os-versions=../calico_versions.yml > pkg/components/calico.go
```

Comment:
I'm wondering if there's a practical way to generate this file from tigera/operator:config/calico_versions.yml.

As it stands there are only relatively minor variations from that file (mostly master vs latest-amd64 but it's not quite a simple swap).

For older branches it's also a little annoying: the operator tends to pin a release version (e.g. v3.28.2), but I guess the intention here is to always use the latest-amd64 tag regardless of what branches are in use?  Might be clearer to swap everything to test-build if that's the case?

Reply:
yeah, the intention is that the version we build is always going to install the `latest` images that were built as part of the CI / commit under test.

I think this will work for release-branches as well. We could change how we build / tag images for CI so that they match the ones expected for the operator that is being pulled in - that would allow us to remove this I think?

Well-worth doing IMO, but I think that's a phase 2 task.